### PR TITLE
[feature-27] Add chat to the GameWindow

### PR DESCRIPTION
**Pull request by lucas-mrq**

### CODE REVIEW

File: src/client/GameWindow.cpp [hunk 1]
```diff
@@ -202,28 +224,23 @@ bool GameWindow::handleGameEvent(sf::Event &event, sf::Vector2i &clickStartingPo
...
-        return true;
-
     case sf::Keyboard::Escape:
         gameEnginePtr->handleQuitMenu(true);
         return true;
 
-    case sf::Keyboard::L:
-        moveMap(clickStartingPoint, {MAP_X_OFFSET, MAP_Y_OFFSET}, true);
+    case sf::Keyboard::Enter:
+        sendMessage();
```
Comment (jard-111) [inline src/client/GameWindow.cpp:236]: even if chat is close ?

File: src/client/GameWindow.cpp [hunk 2]
```diff
@@ -202,28 +224,23 @@ bool GameWindow::handleGameEvent(sf::Event &event, sf::Vector2i &clickStartingPo
...
+
+        //this line should be used when the client receive a message
+        //chatBox->updateChat("00:00", "Username", chatBox->message);  //line to be delete
```
Comment (jard-111) [inline src/client/GameWindow.cpp:239]: delete it useless

File: src/client/GameWindow.cpp (hunk 1 above)
Comment (lucas-mrq) [reply src/client/GameWindow.cpp:236]: It should not, I will correct it

File: src/client/GameWindow.cpp (hunk 2 above)
Comment (lucas-mrq) [reply src/client/GameWindow.cpp:239]: Done

File: src/client/GameWindow.cpp (hunk 1 above)
Comment (lucas-mrq) [reply src/client/GameWindow.cpp:236]: Done

**Review by jard-111 — DISMISSED** [review]
add an empty line at the end of files !

File: src/client/GameWindow.cpp [hunk 3]
```diff
@@ -400,47 +439,73 @@ bool GameWindow::priorityCardClickAction(sf::Vector2i clickPosition)
...
+    if (!*moveMode)
     {
-        for (unsigned j = 0; j < mapTexture.getSize(); j++)
-        { 
-            if (gameEnginePtr->intersectPointRect(clickPosition, mapTexture.getSprite(j).getGlobalBounds()))
+        if (priorityCardClickAction(clickPosition)) {
+            return false;
+        }
+
+        for (auto &mapTexture : mapTextureToDisplay)
```
Comment (NiskuT) [inline src/client/GameWindow.cpp:455]: At some point we will really need a function for that, the `if (!*moveMode)` add another level of indentation

File: src/client/TextureDisplayer.cpp [hunk 4]
```diff
@@ -137,6 +137,27 @@ void TextureDisplayer::setHudSpritePosition(float scale, int windowLength, int w
         break;
     }
 
+    case HudTextureType::chat:
+    {
+        xPos = 20;
+        yPos = 585;
```
Comment (NiskuT) [inline src/client/TextureDisplayer.cpp:143]: magic

**Review by NiskuT — APPROVED** [review]

**Pull request merged**